### ISSUE TITLE
Optional source maps in production build

### ISSUE THREAD

**Issue by martinleopold** [StudioProcess/rvp#172]
To support debugging, it would be nice to have source maps in the production build.

Can we just do this using `npm run build` instead of `npm run build:prod`? Are there any drawbacks to this, like offline service workers not being used etc?

**Comment by martinleopold** [comment]
Just merged the PR #178 but I do get an error.

I did:
```
git checkout master
npm run reinstall
npm run build:prod
```
I am getting:
```
ERROR in ./node_modules/video.js/es5/utils/url.js
Module build failed (from ./node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js):
Error: Debug Failure. False expression.
    at getJSDocTags (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13197:22)
    at getFirstJSDocTag (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13205:24)
    at Object.getJSDocType (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13173:19)
    at Object.getEffectiveTypeAnnotationNode (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:11610:60)
    at tryGetTypeFromEffectiveTypeNode (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30988:31)
    at getTypeForVariableLikeDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30689:32)
    at getWidenedTypeForVariableLikeDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30961:24)
    at getTypeOfVariableOrParameterOrProperty (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31062:28)
    at getTypeOfSymbol (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31242:24)
    at checkVariableLikeDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:47828:41)
    at checkParameter (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45756:13)
    at Object.forEach (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:1792:30)
    at checkSignatureDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45895:16)
    at checkFunctionExpressionOrObjectLiteralMethod (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:44645:21)
    at checkExpressionWorker (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45681:28)
    at checkExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45609:42)
ERROR in ./node_modules/video.js/es5/utils/log.js
Module build failed (from ./node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js):
Error: Debug Failure. False expression.
    at getJSDocTags (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13197:22)
    at getFirstJSDocTag (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13205:24)
    at Object.getJSDocType (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13173:19)
    at Object.getEffectiveTypeAnnotationNode (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:11610:60)
    at tryGetTypeFromEffectiveTypeNode (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30988:31)
    at getTypeForVariableLikeDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30689:32)
    at getWidenedTypeForVariableLikeDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30961:24)
    at getTypeOfVariableOrParameterOrProperty (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31062:28)
    at getTypeOfSymbol (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31242:24)
    at checkVariableLikeDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:47828:41)
    at checkParameter (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45756:13)
    at Object.forEach (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:1792:30)
    at checkSignatureDeclaration (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45895:16)
    at checkFunctionExpressionOrObjectLiteralMethod (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:44645:21)
    at checkExpressionWorker (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45681:28)
    at checkExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45609:42)
ERROR in ./node_modules/video.js/es5/utils/fn.js
Module build failed (from ./node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js):
Error: Debug Failure. False expression.
    at getJSDocTags (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13197:22)
    at getFirstJSDocTag (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13205:24)
    at Object.getJSDocType (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13173:19)
    at getContextualSignature (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:41002:32)
    at checkFunctionExpressionOrObjectLiteralMethod (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:44621:43)
    at checkExpressionWorker (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45681:28)
    at checkExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45609:42)
    at checkExpressionCached (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45457:38)
    at _loop_3 (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30802:56)
    at getWidenedTypeFromJSSpecialPropertyDeclarations (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30837:31)
    at getTypeOfVariableOrParameterOrProperty (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31032:28)
    at getTypeOfSymbol (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31242:24)
    at checkPropertyAccessExpressionOrQualifiedName (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:42338:53)
    at checkPropertyAccessExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:42293:20)
    at checkExpressionWorker (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45663:28)
    at checkExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45609:42)
ERROR in ./node_modules/video.js/es5/utils/stylesheet.js
Module build failed (from ./node_modules/@angular-devkit/build-optimizer/src/build-optimizer/webpack-loader.js):
Error: Debug Failure. False expression.
    at getJSDocTags (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13197:22)
    at getFirstJSDocTag (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13205:24)
    at Object.getJSDocType (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:13173:19)
    at getContextualSignature (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:41002:32)
    at checkFunctionExpressionOrObjectLiteralMethod (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:44621:43)
    at checkExpressionWorker (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45681:28)
    at checkExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45609:42)
    at checkExpressionCached (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45457:38)
    at _loop_3 (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30802:56)
    at getWidenedTypeFromJSSpecialPropertyDeclarations (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:30837:31)
    at getTypeOfVariableOrParameterOrProperty (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31032:28)
    at getTypeOfSymbol (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:31242:24)
    at checkPropertyAccessExpressionOrQualifiedName (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:42338:53)
    at checkPropertyAccessExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:42293:20)
    at checkExpressionWorker (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45663:28)
    at checkExpression (/Users/ml/Documents/code/rvp/node_modules/@angular-devkit/build-optimizer/node_modules/typescript/lib/typescript.js:45609:42)
```

**Comment by dinony** [comment]
Seems to be an issue with the source map builder and the video.js file. An option would be to use the older video.js release again.
